landlab/components/groundwater/dupuit_percolator.py

```diff
@@ -320,6 +321,12 @@ def __init__(
             This parameter is only used with ``run_with_adaptive_time_step_solver``
             and must be greater than zero.
             Default = 0.8
+        k_func: function
+            If the lateral hydraulic conductivity is not vertically homogenous
+            it is necessary to modify the effective hydraulic conductivity based
+            on the position of the water table. The function should take a Landlab
+            ModelGrid. Any other arguments should be given as keyword arguments.
```

Comment:
1. Makes sense. My general intent was that it should return an array at link. I think with the way it is set up, there's no reason that it can't return a single value, though I can't think of scenario where you would want to change everything uniformly.
2. Do you think I should just leave the line about keyword arguments out then? I thought a lot about whether there was a way to add additional arguments in a general kind of way, but I think the way I have it now is a decent solution, at least for my needs.

Reply:
1) If you think it should always return something spatially variable (which does make sense), then I'd keep as is. 

2) I also thought a bit about ways to pass around keyword arguments and came up short.... I'll keep thinking.

Reply:
I decided to keep the test in the setter to check that it's an array at link. I think this makes the most sense.

Reply:
1. I would suggest removing the part about keyword arguments.
1. If I understand correctly, if a user would like to use a function with keyword arguments, perhaps they could use the *partial* function to create a new function with the appropriate keywords set to what they like. As an example, if you wanted *k_func* to return an array of uniformly-distributed random numbers at links, you could do this with the *random* function from *landlab.values* and *partial*,
```python
from functools import partial
from landlab.values import random

k_func = partial(random, "hydraulic_conductivity", "at="link")
```
You can then call *k_func(grid)* and get an array of random numbers between 0.0 and 1.0 at link.

Reply:
Thanks for the note @mcflugen! I haven't used partial before, but looking at what it does, that makes a lot of sense. I removed the language about keyword arguments in one of the previous commits.